metrics/lib/common.bash

```diff
@@ -273,9 +286,13 @@ wait_ksm_settle(){
 
 	oldscan=$(cat /sys/kernel/mm/ksm/full_scans)
 
+	# Wait some time for KSM to kick in to avoid early dismissal
+	local initial_wait=5
+	sleep ${initial_wait}
+
```

Comment:
I'm not sure I understand this. We're giving KSM 5 extra seconds to start but then waiting for 5 less seconds for it to settle? Can't we avoid the `sleep` and just increase the timeout in the loop below (or use a timed wait here for KSM to trigger before entering the loop?)

Reply:
So without the proposed change, this is run inside the loop:
```sh
newpages=$(cat "${KSM_PAGES_SHARED}")    
[[ "$newpages" -eq 0 ]] && echo "No need to wait for KSM to settle" && return
```

and if KSM hasn't started its work right there and then, this will terminate.
> Can't we [...] use a timed wait here for KSM to trigger before entering the loop?

Isn't that what I'm doing with the `sleep`?

Reply:
Unless I'm missing something, ... not really, no ;) You're just sleeping in the "hope" that KSM will start in 5 seconds right? If it takes 6 seconds, the loop below will fail. If it starts in 1 second, the test has to wait a further 4 seconds before it can move to the for loop below.

What I'm saying is can't you use an addional for loop like the one below to poll for some (reasonably large amount of) time for KSM to "start"? Once we know it's started, we can then fall into the "wait for KSM to settle" loop below. By having two loops:

- you can make the logic clearer
- you don't penalise systems where KSM starts quickly as you can jump out of the first loop and hit the second where you wait for KSM to settle.

Reply:
Ah, now I get what you meant :) Let me get back.

Reply:
Updated.